specs/waku/v2/waku-relay.md

```diff
@@ -83,13 +114,30 @@ The `subscribe` field MUST contain a boolean, where 1 means subscribe and 0 mean
+Note that removing identifiable information from messages cannot lead to perfect anonymity. The direct connections of a publisher might be able to figure out which `Message`s belong to that publisher by analyzing its traffic. The possibility of such inference may get higher when the `data` field is also not encrypted. <!-- TODO: more investigation on traffic analysis attacks and their success probability-->
+
+## Future work
+
+- **Confidentiality**: The current version of the `relay` protocol does not address confidentiality, however, it is achievable by performing encryption on top of the `data` field of `Message`s. More details on enabling encryption mode are provided in [Libp2p PubSub specs](https://github.com/libp2p/specs/tree/master/pubsub#the-topic-descriptor) as part of the `TopicDescriptor` messages.
+  
+- **Integrity** and  **Authenticity**: These properties are typically addressed through digital signatures or Message Authentication Code (MAC) schemes, however, the usage of digital signatures (where each signature is bound to a particular peer) contradicts with the anonymity requirement (messages signed under a certain signature key are verifiable by a verification key that is bound to a particular peer).  As such, integrity and authenticity are missing features in the `relay` protocol in the interest of anonymity. To fill this gap, we propose the integration of advanced signature schemes like group signatures to enable authenticity, integrity, and anonymity simultaneously. In a group signature scheme, a member of a group can anonymously sign a message on behalf of the group as such the true signer is indistinguishable from other group members. <!-- TODO: shall I add a reference for group signatures?-->
+
+- **Spam resistant**: This feature is not yet supported by the `relay` protocol, however, a PoC is in progress regarding the utilization of Rate Limiting Nullifiers to protect against spamming and spammers. More details can be found in [Waku RLN Relay](https://github.com/vacp2p/specs/blob/master/specs/waku/v2/waku-rln-relay.md). 
+  <!-- TODO: Maybe mentioning Peer scoring and PoW-->
+  
+- **Topic Subscriber Anonymity:** Concealing the link between a subscriber and its subscribed topic ids can not be addressed at the `relay` protocol since the exposure of this information is key to maintain a topic mesh. In specific,  subscribers can only participate in a topic mesh by getting connected to other subscribers of the same topic id hence disclosing their interest in that topic id at least to a subset of other subscribers (more details on this can be found in [libp2p pubsub documentation](https://docs.libp2p.io/concepts/publish-subscribe/)). As such, this information inevitably gets exposed to the topic mesh. However, upper-level protocols can implement [Partitioned topics](https://specs.status.im/spec/10#partitioned-topic) to provide K-anonymity for peers' subscribed topic ids.
 ## Changelog
 
 ### 2.0.0-beta2
```

Comment:
Hm, something weird happened here. We are already on beta2.

Would be good to add a one-liner to `### Next`, with something like "added initial threat model, security analysis ~". Then for beta3 we just change `Next` to beta3 and it is easier to see how spec has evolved over time.